common/Audit.go

```diff
@@ -0,0 +1,98 @@
+package common
+
+import "github.com/google/uuid"
+
+type AuditRecord struct {
+	auditMap map[uuid.UUID][]int
+	duration int
+	cost     int
+	// reliability float64
+}
+
+func NewAuditRecord(duration int) *AuditRecord {
+	cost := calculateCost(duration)
+
+	return &AuditRecord{
+		auditMap: make(map[uuid.UUID][]int),
+		duration: duration,
+		cost:     cost,
+	}
+}
+
+// Getters
+func (a *AuditRecord) GetAuditMap() map[uuid.UUID][]int {
+	return a.auditMap
+}
+
+func (a *AuditRecord) GetAuditDuration() int {
+	return a.duration
+}
+
+func (a *AuditRecord) GetAuditCost() int {
+	return a.cost
+}
+
+// Setters
+func (a *AuditRecord) SetAuditDuration(duration int) {
+	a.duration, a.cost = duration, calculateCost(duration)
+}
+
+// Implement a more sophisticated cost calculation if needed, could compound with reliability if implemented
+func calculateCost(duration int) int {
+	return duration
+}
+
+// Get the number of infractions in the last n rounds, given by the quality of the audit
+func (a *AuditRecord) GetAllInfractions(agentId uuid.UUID) int {
+	infractions := 0
+	records := a.auditMap[agentId]
+
+	history := len(records)
+	if history > a.duration {
+		history = a.duration
+	}
```

Comment:
```suggestion
history := math.Min(len(records), a.duration)
```

Nitpick :) would need `import "math"` up top though